### ISSUE TITLE
implement balance storage in enclave

### ISSUE THREAD

**Issue by brenzi** [Polkadex-Substrate/polkadexTEE-worker#3]
all balances shall be stored in a HashMap:
```
map (tokenID, AccountID) -> (balance free, balance reserved)
```

To interact with the storage, an interface should be provided, offering the following interactions:
* read balance of a specific account and tokenID
* Mutate balances of a specific account and tokenID

The balances hashmap shall be writable by
* chain relay: upon a `ocex::deposit` or `ocex::withdraw` extrinsic
* GW: settle matched orders
* RPC: upon withdraw direct call

as these are in different threads, the balance storage should provide an atomic pointer with mutex

**Comment by Gauthamastro** [comment]
How do we sync balance on enclave startup?

**Comment by brenzi** [comment]
@Gauthamastro there is no special need for syncing balances at startup. We're only running a single instance. If that instance has downtime, no balance changes will occur. Whatever happens onchain will be synched with the chain relay once the worker is up.
Therefore we just need to load the last state (snapshot) at startup.

We will need to look deeper into this after the PoC

**Comment by Gauthamastro** [comment]
> @Gauthamastro there is no special need for syncing balances at startup. We're only running a single instance. If that instance has downtime, no balance changes will occur. Whatever happens onchain will be synched with the chain relay once the worker is up.
> Therefore we just need to load the last state (snapshot) at startup.
> 
> We will need to look deeper into this after the PoC


Agreed, I believe we can use similar logic that we use for orderbook mirror's permanent storage

**Comment by Gauthamastro** [comment]
Merged in #48
